plugins/modules/meraki_ms_access_polices.py

```diff
@@ -0,0 +1,553 @@
+                "systems_management_enrollment": meraki.params[
+                    "systems_management_enrollment"
+                ],
+                "guestVlanId": meraki.params["guest_vlan"],
+                "urlRedirectWalledGardenEnabled": False,
+            }
+            if meraki.params["number"]:
+                payload.update(
+                    {
+                        "access_policy_number": meraki.params["number"],
+                    }
+                )
+
+            response = meraki.request(
+                path, method=method, payload=json.dumps(payload)
+            )
+            meraki.result["data"] = response
+        elif meraki.params["auth_method"] == "my RADIUS server":
+            payload = {
+                "name": meraki.params["name"],
+                "radiusServers": meraki.params["radius_servers"],
+                "radiusTestingEnabled": meraki.params["radius_testing"],
+                "radiusGroupAttribute": meraki.params[
+                    "radius_attribute_group_policy_name"
+                ],
+                "radius": {
+                    "criticalAuth": {
+                        "dataVlanId": convert_vlan_id(
+                            meraki.params["data_vlan_id"]
+                        ),
+                        "voiceVlanId": convert_vlan_id(
+                            meraki.params["voice_vlan_id"]
+                        ),
+                        "suspendPortBounce": meraki.params[
+                            "suspend_port_bounce"
+                        ],
+                    },
+                    "failedAuthVlanId": None,
+                    "reAuthenticationInterval": None,
+                },
+                "radiusCoaSupportEnabled": meraki.params["radius_coa_enabled"],
+                "hostMode": meraki.params["host_mode"],
+                "accessPolicyType": meraki.params["access_policy_type"],
+                "guestVlanId": meraki.params["guest_vlan"],
+                "voiceVlanClients": meraki.params["voice_vlan_clients"],
+                "urlRedirectWalledGardenEnabled": False,
+                "radiusAccountingEnabled": meraki.params[
+                    "radius_accounting_enabled"
+                ],
+                "systems_management_enrollment": meraki.params[
+                    "systems_management_enrollment"
+                ],
+            }
+            if meraki.params["radius_accounting_enabled"]:
+                payload.update(
+                    {
+                        "radiusAccountingServers": meraki.params[
+                            "radius_accounting_servers"
+                        ],
+                    }
+                )
+            if meraki.params["number"]:
+                payload.update(
+                    {
+                        "access_policy_number": meraki.params["number"],
+                    }
+                )
+
+            response = meraki.request(
```

Comment:
I mentioned this in the integration test. But the modules should always check for changes so it is idempotent. https://github.com/CiscoDevNet/ansible-meraki/blob/2d955b941dfbe39c86204930b6c92d8138ef501d/plugins/modules/meraki_mx_nat.py#L600 is an example of how it's done. Admittedly, this is one of the more error prone areas of code so let me know if you need assistance. It should happen prior to any request being sent.